pkg/controller/template/template_controller.go

```diff
@@ -341,6 +355,37 @@ func (ctrl *Controller) deleteMachineConfig(obj interface{}) {
 	ctrl.enqueueControllerConfig(cfg)
 }
 
+func (ctrl *Controller) addStorage(obj interface{}) {
+	storage := obj.(*operatorv1.Storage)
+	glog.V(4).Infof("Adding Storage %s", storage.Name)
+	ctrl.enqueueController()
+}
+
+func (ctrl *Controller) updateStorage(old, cur interface{}) {
+	oldStorage := old.(*operatorv1.Storage)
+	glog.V(4).Infof("Updating Storage %s", oldStorage.Name)
+	ctrl.enqueueController()
+}
```

Comment:
we are seeing an issue where cluster installations are not happening properly and the master nodes are coming up with a different configuration than expected. it appears like the Storage object is `nil` when the master machineconfig templates are created, and i wonder if this function needs a change to ensure they get updated?

Reply:
The issue I believe is with the bootstrap, see my comment about decoding and the fact we don't have the infra object